retrievalmarket/impl/provider.go

```diff
@@ -327,6 +370,31 @@ func (p *Provider) HandleQueryStream(stream rmnet.RetrievalQueryStream) {
 	}
 }
 
+func (p *Provider) GetDynamicAsk(ctx context.Context, input retrievalmarket.PricingInput, storageDeals []abi.DealID) (retrievalmarket.Ask, error) {
```

Comment:
Could we add a comment to GetDynamicAsk explaining the difference between this function and GetAsk